addons/survey/controllers/main.py

```diff
@@ -472,6 +472,50 @@ def survey_begin(self, survey_token, answer_token, **post):
         answer_sudo._mark_in_progress()
         return self._prepare_question_html(survey_sudo, answer_sudo, **post)
 
+    @http.route('/survey/<string:survey_token>/<string:answer_token>/check_answers', type='json', auth='public', website=True)
+    def survey_check_answers(self, survey_token, answer_token, **post):
+        access_data = self._get_access_data(survey_token, answer_token, ensure_token=True)
+        if access_data['validity_code'] is not True:
+            return {'error': access_data['validity_code']}
+        survey_sudo, answer_sudo = access_data['survey_sudo'], access_data['answer_sudo']
+
+        if answer_sudo.state == 'done':
+            return {'error': 'unauthorized'}
+        if survey_sudo.scoring_type != 'scoring_with_answers_after_page':
+            return {'error': 'unauthorized'}
+
+        user_answers = post.get('user_answers')
+        if not user_answers:
+            return False
```

Comment:
Wouldn't we be more efficient if we first checked for this before calling `_get_access_data`? (and maybe return an error of some kind?)